src/web/mjs/connectors/templates/SpeedBinb.mjs

```diff
@@ -160,22 +164,26 @@ export default class SpeedBinb extends Connector {
     _getPageList_v016130( chapterID, apiURL ) {
         let cid = new URL( chapterID, this.url ).searchParams.get( 'cid' );
         let sharingKey = this._tt( cid );
-        let uri = new URL( apiURL, this.url );
+        let uri;
+        if ( this.url.includes("hobbyjapan") ) 
+            uri = new URL( apiURL, chapterID.substr(0, chapterID.indexOf('?')) + '/' ); //hj uses chapterID instead of this.url
+        else 
+            uri = new URL( apiURL, this.url );
         uri.searchParams.set( 'cid', cid );
         uri.searchParams.set( 'dmytime', Date.now() );
         uri.searchParams.set( 'k', sharingKey );
         let request = new Request( uri.href, this.requestOptions );
         return fetch( request )
             .then( response => response.json() )
             .then( data => {
-                return this._getPageLinks_v016130( data.items[0], sharingKey );
+                return this._getPageLinks_v016130( data.items[0], sharingKey, chapterID );
             } );
     }
 
     /**
      *
      */
-    _getPageLinks_v016130( configuration, sharingKey ) {
+    _getPageLinks_v016130( configuration, sharingKey, chapterID ) {
```

Comment:
Do not change this method, instead adjust the configuration parameter when this method is called